avocado/utils/podman.py

```diff
@@ -0,0 +1,104 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2021 Red Hat Inc.
+# Authors : Beraldo Leal <bleal@redhat.com>
+
+"""
+This module provides an basic API for interacting with podman.
+
+This module it was designed to be executed in async mode. Remember this when
+consuming this API.
+"""
+
+import json
+import logging
+import os
+
+from asyncio import create_subprocess_exec, subprocess
+
+
+LOG = logging.getLogger(__name__)
+
+
+class PodmanException(Exception):
+    pass
+
+
+class Podman:
+    def __init__(self, podman_bin):
+        if not os.path.exists(podman_bin):
+            msg = f"Podman binary {podman_bin} is not available on the system."
+            raise PodmanException(msg)
+
+        self.podman_bin = podman_bin
+
+    async def execute(self, *args):
+        """Execute a command and return the returncode, stdout and stderr.
+
+        :param *args: Variable length argument list to be used as argument
+                      during execution.
+        :rtype: tuple with returncode, stdout and stderr.
+        """
+        try:
+            proc = await create_subprocess_exec(self.podman_bin,
+                                                *args,
+                                                stdout=subprocess.PIPE,
+                                                stderr=subprocess.PIPE)
+            stdout, stderr = await proc.communicate()
+        except (FileNotFoundError, PermissionError) as ex:
+            # Since this method is also used by other methods, let's
+            # log here as well.
+            msg = "Could not execute the command."
+            LOG.error("%s: %s", msg, str(ex))
+            raise PodmanException(msg) from ex
+
+        if proc.returncode != 0:
+            msg = f"Could not execute the command: {proc.returncode}:{stderr}."
+            LOG.error(msg)
+            raise PodmanException(msg)
+
+        return proc.returncode, stdout, stderr
+
+    async def get_python_version(self, image):
+        """Return the current Python version installed in an image.
+
+        :param str image: Image name. i.e: 'fedora:33'.
+        :rtype: tuple with both: major and minor numbers. Or None (in case of
+                error).
+        """
+
+        entrypoint = json.dumps(["/usr/bin/env", "python3", "-c",
```

Comment:
IIUC, you're using env simply because the `python3` binary may be in different places depending on the image, right?  I have the feeling that it'd be good to also return the `sys.executable` location, so that it can be used directly later on.